### PR TITLE
Implementation Status Tables

### PR DESCRIPTION
**Pull request by diasdavid**
> WIP

Need to update the http-api tables from the cli as Apiary is vastly out of date.

### CODE REVIEW

File: IMPLEMENTATION_STATUS.md [hunk 1]
```diff
@@ -0,0 +1,588 @@
...
+
+#### CLI
+
+#### HTTP
+
+| Endpoint                                     | Go Impl       | JS Impl       |
+| -------------------------------------------- | :-----------: | :-----------: |
+| [`GET /api/v0/bootstrap/add`]                | :green_apple: | :green_apple: |
+| [`GET /api/v0/bootstrap/list`]               | :green_apple: | :green_apple: |
+| [`GET /api/v0/bootstrap/rm`]                 | :green_apple: | :green_apple: |
```
Comment (Kubuxu) [inline IMPLEMENTATION_STATUS.md]: rm has `--all` option, does js cover it?

File: IMPLEMENTATION_STATUS.md [hunk 2]
```diff
@@ -0,0 +1,588 @@
...
+
+## Bootstrap
+
+#### CLI
+
+#### HTTP
+
+| Endpoint                                     | Go Impl       | JS Impl       |
+| -------------------------------------------- | :-----------: | :-----------: |
+| [`GET /api/v0/bootstrap/add`]                | :green_apple: | :green_apple: |
```
Comment (Kubuxu) [inline IMPLEMENTATION_STATUS.md]: add has `--default` option, does js cover it?

File: IMPLEMENTATION_STATUS.md [hunk 3]
```diff
@@ -0,0 +1,588 @@
...
+#### CLI
+
+#### HTTP
+
+| Endpoint                                     | Go Impl       | JS Impl       |
+| -------------------------------------------- | :-----------: | :-----------: |
+| [`POST /api/v0/config`]                      | :green_apple: | :chestnut:    |
+| [`POST /api/v0/config/replace`]              | :green_apple: | :chestnut:    |
+| [`GET /api/v0/config/show`]                  | :green_apple: | :chestnut:    |
+| [`POST /api/v0/log/level`]                   | :green_apple: | :chestnut:    |
```
Comment (Kubuxu) [inline IMPLEMENTATION_STATUS.md]: Is `log` part of `config` section on purpose?

File: IMPLEMENTATION_STATUS.md [hunk 4]
```diff
@@ -0,0 +1,588 @@
...
+| **`POST /api/v0/add`**                       | :green_apple: | :lemon:       |
+|     `arg=`                                   | :green_apple: | :chestnut:    |
+|     `H=`                                     | :green_apple: | :chestnut:    |
+|     `n=`                                     | :green_apple: | :chestnut:    |
+|     `p=`                                     | :green_apple: | :chestnut:    |
+|     `pin=`                                   | :green_apple: | :chestnut:    |
+|     `r=`                                     | :green_apple: | :chestnut:    |
+|     `s=`                                     | :green_apple: | :chestnut:    |
+|     `t=`                                     | :green_apple: | :chestnut:    |
+|     `w=`                                     | :green_apple: | :chestnut:    |
```
Comment (Kubuxu) [inline IMPLEMENTATION_STATUS.md]: I think it might be better using full option names here.

File: IMPLEMENTATION_STATUS.md [hunk 5]
```diff
@@ -0,0 +1,588 @@
...
+
+
+| Endpoint                                     | Go Impl       | JS Impl       |
+| -------------------------------------------- | :-----------: | :-----------: |
+| **`GET /api/v0/ping`**                       | :green_apple: | :chestnut:    |
+|     `arg=`                                   | :green_apple: | :chestnut:    |
+|     `count=,n=`                              | :green_apple: | :chestnut:    |
+| **`GET /api/v0/update`**                     | :chestnut:    | :chestnut:    |
+| **`GET /api/v0/version`**                    | :green_apple: | :chestnut:    |
+| **`GET /api/v0/tour`**                       | :green_apple: | :chestnut:    |
```
Comment (Kubuxu) [inline IMPLEMENTATION_STATUS.md]: Tour was removed.

File: IMPLEMENTATION_STATUS.md [hunk 6]
```diff
@@ -0,0 +1,588 @@
...
+#### CLI
+
+#### HTTP
+
+| Endpoint                                     | Go Impl       | JS Impl       |
+| -------------------------------------------- | :-----------: | :-----------: |
+| [`GET /api/v0/diag/cmds`]                    | :green_apple: | :chestnut:    |
+| [`GET /api/v0/diag/cmds/clear`]              | :green_apple: | :chestnut:    |
+| [`GET /api/v0/diag/cmds/set-time`]           | :green_apple: | :chestnut:    |
+| [`GET /api/v0/net`]                          | :green_apple: | :chestnut:    |
```
Comment (Kubuxu) [inline IMPLEMENTATION_STATUS.md]: diag net was removed.

File: IMPLEMENTATION_STATUS.md [hunk 7]
```diff
@@ -0,0 +1,588 @@
...
+|     `recursive=`                             | :green_apple: | :tomato:      |
+| **`GET /api/v0/refs`**                       | :green_apple: | :tomato:      |
+|     `arg=`                                   | :green_apple: | :tomato:      |
+|     `format=<src>`                           | :green_apple: | :tomato:      |
+|     `format=<dst>`                           | :green_apple: | :tomato:      |
+|     `format=<linkname>`                      | :green_apple: | :tomato:      |
+|     `edges=,e=`                              | :green_apple: | :tomato:      |
+|     `unique=,u=`                             | :green_apple: | :tomato:      |
+|     `recursive=,r=`                          | :green_apple: | :tomato:      |
+| **`GET /api/v0//refs/local`**                | :green_apple: | :tomato:      |
```
Comment (Kubuxu) [inline IMPLEMENTATION_STATUS.md]: I don't think refs should be part of `pinning`.

File: IMPLEMENTATION_STATUS.md (hunk 1 above)
Comment (diasdavid) [inline IMPLEMENTATION_STATUS.md]: @Kubuxu see above _"WIP"_ means that I'm not finished. The Apiary documentation was out of date

File: IMPLEMENTATION_STATUS.md (hunk 6 above)
Comment (diasdavid) [inline IMPLEMENTATION_STATUS.md]: Mind updating https://github.com/ipfs/http-api-spec ?

File: IMPLEMENTATION_STATUS.md (hunk 5 above)
Comment (diasdavid) [inline IMPLEMENTATION_STATUS.md]: Seems not, am I see it wrong?
```
» ipfs tour

Tour 0.0 - Hello Mars


        check things work


 ⭠ implementation-status ⮀ ~/code/ipfs ⮀
» ipfs tour --help
USAGE
  ipfs tour [<id>] - Provide an introduction to IPFS.

SYNOPSIS
  ipfs tour [--] [<id>]

ARGUMENTS

  [<id>] - The id of the topic you would like to tour.

DESCRIPTION

  This is a tour that takes you through various IPFS concepts,
  features, and tools to make sure you get up to speed with
  IPFS very quickly. To start, run:

      ipfs tour

SUBCOMMANDS
  ipfs tour list    - Show a list of IPFS Tour topics.
  ipfs tour next    - Show the next IPFS Tour topic.
  ipfs tour restart - Restart the IPFS Tour.

  Use 'ipfs tour <subcmd> --help' for more information about each command.
```

File: IMPLEMENTATION_STATUS.md (hunk 6 above)
Comment (diasdavid) [inline IMPLEMENTATION_STATUS.md]: I went ahead and:
- updated https://github.com/ipfs/http-api-docs
- published the website with a new version of https://github.com/ipfs/website/blob/master/content/docs/api.md
- added a notice tot he old http-api-spec https://github.com/ipfs/http-api-spec#psa---this-specification-is-out-of-date-please-use-api-docs-for-a-up-to-date-reference

File: IMPLEMENTATION_STATUS.md (hunk 7 above)
Comment (diasdavid) [inline IMPLEMENTATION_STATUS.md]: They are often used together. I can break it into a separate table

**Comment by diasdavid** [comment]
# Status Update

Had to go through all of our multiple versions of [HTTP API docs and give a summer clean up](https://github.com/ipfs/ipfs/pull/264#discussion_r134668760). tl;dr; There is a lot of work to do.

## Tracking progress

- [ ] Bitswap
  - [ ] review HTTP endpoints based on http-api-docs (and not http-api-spec)
  - [ ] Add table for CLI
  - [ ] Review and update status in all the endpoints
- [ ] Block
  - [ ] review HTTP endpoints based on http-api-docs (and not http-api-spec)
  - [ ] Add table for CLI
  - [ ] Review and update status in all the endpoints
- [ ] Bootstrap
  - [ ] review HTTP endpoints based on http-api-docs (and not http-api-spec)
  - [ ] Add table for CLI
  - [ ] Review and update status in all the endpoints
- [ ] Config
  - [ ] review HTTP endpoints based on http-api-docs (and not http-api-spec)
  - [ ] Add table for CLI
  - [ ] Review and update status in all the endpoints
- [ ] DAG
  - [ ] list out HTTP endpoints based on http-api-docs (and not http-api-spec)
  - [ ] Add table for CLI
  - [ ] Review and update status in all the endpoints
- [ ] Diagnostics
  - [ ] review HTTP endpoints based on http-api-docs (and not http-api-spec)
  - [ ] Add table for CLI
  - [ ] Review and update status in all the endpoints
- [ ] DHT
  - [ ] review HTTP endpoints based on http-api-docs (and not http-api-spec)
  - [ ] Add table for CLI
  - [ ] Review and update status in all the endpoints
- [ ] Files
  - [ ] review HTTP endpoints based on http-api-docs (and not http-api-spec)
  - [ ] Add table for CLI
  - [ ] Review and update status in all the endpoints
- [ ] Miscellaneous
  - [ ] review HTTP endpoints based on http-api-docs (and not http-api-spec)
  - [ ] Add table for CLI
  - [ ] Review and update status in all the endpoints
- [ ] Naming
  - [ ] review HTTP endpoints based on http-api-docs (and not http-api-spec)
  - [ ] Add table for CLI
  - [ ] Review and update status in all the endpoints
- [ ] Object
  - [ ] review HTTP endpoints based on http-api-docs (and not http-api-spec)
  - [ ] Add table for CLI
  - [ ] Review and update status in all the endpoints
- [ ] Pining
  - [ ] review HTTP endpoints based on http-api-docs (and not http-api-spec)
  - [ ] Add table for CLI
- [ ] PubSub
  - [ ] list out HTTP endpoints based on http-api-docs (and not http-api-spec)
  - [ ] Add table for CLI
  - [ ] Review and update status in all the endpoints
- [ ] Repo
  - [ ] review HTTP endpoints based on http-api-docs (and not http-api-spec)
  - [ ] Add table for CLI
  - [ ] Review and update status in all the endpoints
- [ ] Swarm
  - [ ] review HTTP endpoints based on http-api-docs (and not http-api-spec)
  - [ ] Add table for CLI
  - [ ] Review and update status in all the endpoints

File: IMPLEMENTATION_STATUS.md (hunk 5 above)
Comment (diasdavid) [inline IMPLEMENTATION_STATUS.md]: Nvm, you removed it from master. It hasn't been released yet.

**Comment by diasdavid** [comment]
Ok did all the ones that are not blocked right now (except PubSub). Missing DHT, Naming, Key Management, FileStore and PubSub.

Going to go ahead and merge this to unblock other folks and then finish it in separate PRs during the rest of the day and tomorrow.

**Pull request merged**